server/base/src/test/java/org/apache/accumulo/server/util/TServerUtilsTest.java

```diff
@@ -233,6 +234,72 @@ public void testStartServerUsedPortWithSearch() throws Exception {
     }
   }
 
+  @SuppressFBWarnings(value = "UNENCRYPTED_SERVER_SOCKET", justification = "socket for testing")
+  @Test
+  public void testStartServerNonDefaultPorts() throws Exception {
```

Comment:
I think I have an idea to simplify the unit testing logic for the port search behavior by putting all the port search logic into a single method that accepts a starting port and a config, and returns a selected port, but I can experiment with that idea after we merge in the functionality you've added.

Reply:
It would be nice to rework this entire unit test to use less static stuff and pass more, but that would be follow on work.  Not something for this PR.

Reply:
I can't remember the entire idea I had to simplify the logic for this test. I looked over the existing test, and it seems fine as-is, but I did open #1815 to clean up some of the static stuff and EasyMock code a bit.